ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -290,6 +190,7 @@ let validate_actions_after_crash ~__context ~self ~value =
 
 (* Used to sanity-check parameters before VM start *)
 let validate_basic_parameters ~__context ~self ~snapshot:x =
+	update_vm_virtual_hardware_platform_version ~__context ~vm:self;
```

Comment:
On second thoughts, I think this might be better in assert_hardware_platform_support.
Both are called from assert_can_boot_here, but assert_hardware_platform_support is called from Xapi_vm_migrate.assert_can_migrate in the cross-pool case which does not call assert_can_boot_here.